test/e2e/clusterclass_changes.go

```diff
@@ -736,15 +815,63 @@ func rebaseClusterClassAndWait(ctx context.Context, input rebaseClusterClassAndW
 
 			return nil
 		}, input.WaitForMachineDeployments...).Should(BeNil())
+
+		Eventually(func(g Gomega) error {
+			// Get MachineDeployment for the current MachineDeploymentTopology.
+			mdList := &clusterv1.MachineDeploymentList{}
+			g.Expect(mgmtClient.List(ctx, mdList, client.InNamespace(input.Cluster.Namespace), client.MatchingLabels{
+				clusterv1.ClusterTopologyMachineDeploymentNameLabel: mdTopology.Name,
+			})).To(Succeed())
+			if len(mdList.Items) != 1 {
+				return errors.Errorf("expected one MachineDeployment for topology %q, but got %d", mdTopology.Name, len(mdList.Items))
+			}
+			md := mdList.Items[0]
+
+			g.Expect(md.Spec.Replicas).To(HaveValue(Equal(md.Status.Replicas)), "Machine deployment does not have expected replicas")
+			g.Expect(md.Spec.Replicas).To(HaveValue(Equal(md.Status.AvailableReplicas)), "Machine deployment does not have expected available replicas")
+			g.Expect(md.Spec.Replicas).To(HaveValue(Equal(md.Status.ReadyReplicas)), "Machine deployment does not have expected ready replicas")
```

Comment:
Can we include these 3 lines in the eventually above? (l.816++)

Reply:
This unfortunately caused flakes, it needed more time. So I separated this instead of changing a global interval or adding a new one.

Reply:
Hm strange. The test seems to only take 3,5min or something (I guess best case?) while this interval is 5m

Reply:
WDYT about setting WorkerMachineCount to 0 in l.181 and getting rid of this here?

The main goal of this test is to ensure the Cluster topology controller rolls out stuff correctly to CP / MD / MP. We intentionally didn't care about the full rollouts up until now